src/reconcile/rec_row.c

```diff
@@ -410,15 +410,17 @@ __wt_rec_row_int(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
         } else {
             /*
              * The transaction ids are cleared after restart. Repack the cell with new validity
-             * information to flush cleared transaction ids. Proxy cells require additional
-             * information in the address cell, be sure to propagate the original fast-truncate
-             * information.
+             * information to flush cleared transaction ids.
              */
             WT_ASSERT_ALWAYS(session, cms.state == WT_CHILD_ORIGINAL,
               "Not propagating the original fast-truncate information");
             __wt_cell_unpack_addr(session, page->dsk, ref->addr, vpack);
+
+            /* The proxy cells of fast truncate pages must be handled in the above flows. */
+            WT_ASSERT_ALWAYS(session, vpack->type != WT_CELL_ADDR_DEL,
+              "Proxy cell is selected with original child image");
+
             if (F_ISSET(vpack, WT_CELL_UNPACK_TIME_WINDOW_CLEARED)) {
-                page_del = vpack->type == WT_CELL_ADDR_DEL ? &vpack->page_del : NULL;
                 __wt_rec_cell_build_addr(session, r, NULL, vpack, WT_RECNO_OOB, page_del);
```

Comment:
Actually, I take that back -- isn't this using page_del without setting it?

(you'll also need to update rec_col_int, btw)

Reply:
Yes. It was using the page_del structure if the cell is a proxy cell. 
Will fix the rec_col_int also.

Reply:
Fixed.